{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-09T01:23:34.414056Z",
     "iopub.status.busy": "2023-11-09T01:23:34.413908Z",
     "iopub.status.idle": "2023-11-09T01:23:34.502664Z",
     "shell.execute_reply": "2023-11-09T01:23:34.502043Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-09T01:23:34.504868Z",
     "iopub.status.busy": "2023-11-09T01:23:34.504698Z",
     "iopub.status.idle": "2023-11-09T03:14:08.634193Z",
     "shell.execute_reply": "2023-11-09T03:14:08.633609Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:23:34.690276: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35129\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:24:57.963065: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:57.974564: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:57.974788: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:57.979046: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:57.979243: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:57.979411: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.050590: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.050840: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.051010: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.051150: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 20:24:58.051385: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:24:58.972428: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.972688: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.972872: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.973083: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.973258: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:24:58.973394: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 20:24:58.973426: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 20:24:58.996444: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 20:24:59.212603: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/recurrent_kernel/Assign' id:3501 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/recurrent_kernel, lstm_21/lstm_cell_21/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:24:59.345606: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:24:59.378798: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31603, 95)\n",
      "Train on 31603 samples, validate on 3526 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:25:03.747886: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:25:07.516720: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 20:25:09.617606: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 20:25:09.636600: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 3.2185"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 20:25:31.384985: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.86390, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 29s 919us/sample - loss: 3.2185 - val_loss: 1.8639\n",
      "Epoch 2/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.7835\n",
      "Epoch 2: val_loss improved from 1.86390 to 1.57416, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 624us/sample - loss: 1.7835 - val_loss: 1.5742\n",
      "Epoch 3/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.5960\n",
      "Epoch 3: val_loss improved from 1.57416 to 1.48858, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 630us/sample - loss: 1.5960 - val_loss: 1.4886\n",
      "Epoch 4/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.5329\n",
      "Epoch 4: val_loss improved from 1.48858 to 1.45059, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 683us/sample - loss: 1.5329 - val_loss: 1.4506\n",
      "Epoch 5/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.5021\n",
      "Epoch 5: val_loss improved from 1.45059 to 1.43077, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.5021 - val_loss: 1.4308\n",
      "Epoch 6/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4894\n",
      "Epoch 6: val_loss improved from 1.43077 to 1.42189, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 644us/sample - loss: 1.4894 - val_loss: 1.4219\n",
      "Epoch 7/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4776\n",
      "Epoch 7: val_loss improved from 1.42189 to 1.41605, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 679us/sample - loss: 1.4776 - val_loss: 1.4160\n",
      "Epoch 8/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4770\n",
      "Epoch 8: val_loss improved from 1.41605 to 1.40875, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 688us/sample - loss: 1.4770 - val_loss: 1.4087\n",
      "Epoch 9/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4710\n",
      "Epoch 9: val_loss improved from 1.40875 to 1.40803, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 1.4710 - val_loss: 1.4080\n",
      "Epoch 10/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4881\n",
      "Epoch 10: val_loss improved from 1.40803 to 1.40202, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 659us/sample - loss: 1.4881 - val_loss: 1.4020\n",
      "Epoch 11/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4617\n",
      "Epoch 11: val_loss improved from 1.40202 to 1.39532, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 626us/sample - loss: 1.4617 - val_loss: 1.3953\n",
      "Epoch 12/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4545\n",
      "Epoch 12: val_loss improved from 1.39532 to 1.39334, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.4545 - val_loss: 1.3933\n",
      "Epoch 13/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4805\n",
      "Epoch 13: val_loss did not improve from 1.39334\n",
      "31603/31603 [==============================] - 20s 618us/sample - loss: 1.4805 - val_loss: 1.3995\n",
      "Epoch 14/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4488\n",
      "Epoch 14: val_loss improved from 1.39334 to 1.38788, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 665us/sample - loss: 1.4488 - val_loss: 1.3879\n",
      "Epoch 15/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4548\n",
      "Epoch 15: val_loss did not improve from 1.38788\n",
      "31603/31603 [==============================] - 21s 673us/sample - loss: 1.4548 - val_loss: 1.3898\n",
      "Epoch 16/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4983\n",
      "Epoch 16: val_loss did not improve from 1.38788\n",
      "31603/31603 [==============================] - 19s 614us/sample - loss: 1.4983 - val_loss: 1.4122\n",
      "Epoch 17/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4764\n",
      "Epoch 17: val_loss did not improve from 1.38788\n",
      "31603/31603 [==============================] - 19s 614us/sample - loss: 1.4764 - val_loss: 1.3926\n",
      "Epoch 18/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4484\n",
      "Epoch 18: val_loss improved from 1.38788 to 1.38302, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 19s 617us/sample - loss: 1.4484 - val_loss: 1.3830\n",
      "Epoch 19/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4879\n",
      "Epoch 19: val_loss did not improve from 1.38302\n",
      "31603/31603 [==============================] - 19s 612us/sample - loss: 1.4879 - val_loss: 1.3976\n",
      "Epoch 20/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4803\n",
      "Epoch 20: val_loss did not improve from 1.38302\n",
      "31603/31603 [==============================] - 19s 612us/sample - loss: 1.4803 - val_loss: 1.3832\n",
      "Epoch 21/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4505\n",
      "Epoch 21: val_loss improved from 1.38302 to 1.37890, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 19s 617us/sample - loss: 1.4505 - val_loss: 1.3789\n",
      "Epoch 22/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4449\n",
      "Epoch 22: val_loss improved from 1.37890 to 1.37805, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 618us/sample - loss: 1.4449 - val_loss: 1.3780\n",
      "Epoch 23/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4407\n",
      "Epoch 23: val_loss did not improve from 1.37805\n",
      "31603/31603 [==============================] - 22s 686us/sample - loss: 1.4407 - val_loss: 1.3803\n",
      "Epoch 24/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4315\n",
      "Epoch 24: val_loss improved from 1.37805 to 1.37402, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 704us/sample - loss: 1.4315 - val_loss: 1.3740\n",
      "Epoch 25/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4221\n",
      "Epoch 25: val_loss improved from 1.37402 to 1.37217, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 1.4221 - val_loss: 1.3722\n",
      "Epoch 26/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4178\n",
      "Epoch 26: val_loss improved from 1.37217 to 1.36742, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 695us/sample - loss: 1.4178 - val_loss: 1.3674\n",
      "Epoch 27/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4132\n",
      "Epoch 27: val_loss improved from 1.36742 to 1.36677, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 679us/sample - loss: 1.4132 - val_loss: 1.3668\n",
      "Epoch 28/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4099\n",
      "Epoch 28: val_loss improved from 1.36677 to 1.36478, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 646us/sample - loss: 1.4099 - val_loss: 1.3648\n",
      "Epoch 29/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4069\n",
      "Epoch 29: val_loss improved from 1.36478 to 1.36120, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 666us/sample - loss: 1.4069 - val_loss: 1.3612\n",
      "Epoch 30/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4044\n",
      "Epoch 30: val_loss improved from 1.36120 to 1.35737, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 19s 614us/sample - loss: 1.4044 - val_loss: 1.3574\n",
      "Epoch 31/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4053\n",
      "Epoch 31: val_loss did not improve from 1.35737\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 1.4053 - val_loss: 1.3591\n",
      "Epoch 32/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3988\n",
      "Epoch 32: val_loss improved from 1.35737 to 1.35545, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 627us/sample - loss: 1.3988 - val_loss: 1.3555\n",
      "Epoch 33/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3978\n",
      "Epoch 33: val_loss improved from 1.35545 to 1.35250, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 662us/sample - loss: 1.3978 - val_loss: 1.3525\n",
      "Epoch 34/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3964\n",
      "Epoch 34: val_loss did not improve from 1.35250\n",
      "31603/31603 [==============================] - 22s 692us/sample - loss: 1.3964 - val_loss: 1.3568\n",
      "Epoch 35/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3939\n",
      "Epoch 35: val_loss improved from 1.35250 to 1.35234, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 1.3939 - val_loss: 1.3523\n",
      "Epoch 36/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3943\n",
      "Epoch 36: val_loss improved from 1.35234 to 1.35027, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 685us/sample - loss: 1.3943 - val_loss: 1.3503\n",
      "Epoch 37/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3909\n",
      "Epoch 37: val_loss improved from 1.35027 to 1.34809, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 658us/sample - loss: 1.3909 - val_loss: 1.3481\n",
      "Epoch 38/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3892\n",
      "Epoch 38: val_loss did not improve from 1.34809\n",
      "31603/31603 [==============================] - 21s 657us/sample - loss: 1.3892 - val_loss: 1.3499\n",
      "Epoch 39/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3858\n",
      "Epoch 39: val_loss improved from 1.34809 to 1.34507, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 629us/sample - loss: 1.3858 - val_loss: 1.3451\n",
      "Epoch 40/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3832\n",
      "Epoch 40: val_loss improved from 1.34507 to 1.34346, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 640us/sample - loss: 1.3832 - val_loss: 1.3435\n",
      "Epoch 41/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3838\n",
      "Epoch 41: val_loss did not improve from 1.34346\n",
      "31603/31603 [==============================] - 22s 694us/sample - loss: 1.3838 - val_loss: 1.3463\n",
      "Epoch 42/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3811\n",
      "Epoch 42: val_loss improved from 1.34346 to 1.34132, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 648us/sample - loss: 1.3811 - val_loss: 1.3413\n",
      "Epoch 43/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3791\n",
      "Epoch 43: val_loss improved from 1.34132 to 1.34067, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3791 - val_loss: 1.3407\n",
      "Epoch 44/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3783\n",
      "Epoch 44: val_loss did not improve from 1.34067\n",
      "31603/31603 [==============================] - 20s 620us/sample - loss: 1.3783 - val_loss: 1.3435\n",
      "Epoch 45/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3757\n",
      "Epoch 45: val_loss did not improve from 1.34067\n",
      "31603/31603 [==============================] - 19s 615us/sample - loss: 1.3757 - val_loss: 1.3410\n",
      "Epoch 46/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3780\n",
      "Epoch 46: val_loss did not improve from 1.34067\n",
      "31603/31603 [==============================] - 20s 620us/sample - loss: 1.3780 - val_loss: 1.3429\n",
      "Epoch 47/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3755\n",
      "Epoch 47: val_loss improved from 1.34067 to 1.33777, saving model to ./checkpoints/unknown_person_baseline_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 698us/sample - loss: 1.3755 - val_loss: 1.3378\n",
      "Epoch 48/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3742\n",
      "Epoch 48: val_loss did not improve from 1.33777\n",
      "31603/31603 [==============================] - 22s 699us/sample - loss: 1.3742 - val_loss: 1.3437\n",
      "Epoch 49/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3728\n",
      "Epoch 49: val_loss did not improve from 1.33777\n",
      "31603/31603 [==============================] - 22s 695us/sample - loss: 1.3728 - val_loss: 1.3385\n",
      "Epoch 50/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3692\n",
      "Epoch 50: val_loss did not improve from 1.33777\n",
      "31603/31603 [==============================] - 22s 699us/sample - loss: 1.3692 - val_loss: 1.3380\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:42:29.621394: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_1_1/bias/Assign' id:24606 op device:{requested: '', assigned: ''} def:{{{node dense_1_1/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_1_1/bias, dense_1_1/bias/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:42:31.388369: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_30_1/lstm_cell_67/bias/m/Assign' id:25712 op device:{requested: '', assigned: ''} def:{{{node lstm_30_1/lstm_cell_67/bias/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_30_1/lstm_cell_67/bias/m, lstm_30_1/lstm_cell_67/bias/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 20:42:32.778953: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1872)\n",
      "(1514, 1872)\n",
      "(1644, 1872)\n",
      "(1764, 1872)\n",
      "(1836, 1872)\n",
      "(1699, 1872)\n",
      "(1369, 1872)\n",
      "(1778, 1872)\n",
      "(1631, 1872)\n",
      "(1704, 1872)\n",
      "(1550, 1872)\n",
      "(1920, 1872)\n",
      "(1739, 1872)\n",
      "(1788, 1872)\n",
      "(1812, 1872)\n",
      "(1788, 1872)\n",
      "(1776, 1872)\n",
      "(946, 1872)\n",
      "(1860, 1872)\n",
      "{1: 6.803251339736257, 2: 3.247182742071646, 4: 8.663663329674456, 5: 7.005909756659408, 6: 7.5099895250083, 8: 8.627500846513978, 9: 3.6126361930434703, 10: 9.453294540001334, 11: 5.855562603421567, 12: 10.0, 13: 7.034103127514378, 17: 9.510408986711216, 19: 9.047342078114264, 21: 9.575442051910283, 22: 1.0, 25: 8.631109669268803, 26: 6.740119218210513, 27: 2.054097028653426, 29: 2.9555066971961743}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2793471/1262401083.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31603 samples, validate on 3526 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:46:15.051803: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 9.9111\n",
      "Epoch 1: val_loss improved from inf to 1.35681, saving model to ./checkpoints/unknown_person_p28_20.h5\n",
      "31603/31603 [==============================] - 28s 886us/sample - loss: 9.9111 - val_loss: 1.3568\n",
      "Epoch 2/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.8482\n",
      "Epoch 2: val_loss did not improve from 1.35681\n",
      "31603/31603 [==============================] - 22s 705us/sample - loss: 9.8482 - val_loss: 1.3634\n",
      "Epoch 3/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.8152\n",
      "Epoch 3: val_loss improved from 1.35681 to 1.34672, saving model to ./checkpoints/unknown_person_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 711us/sample - loss: 9.8152 - val_loss: 1.3467\n",
      "Epoch 4/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.7551\n",
      "Epoch 4: val_loss did not improve from 1.34672\n",
      "31603/31603 [==============================] - 21s 665us/sample - loss: 9.7551 - val_loss: 1.3504\n",
      "Epoch 5/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.7218\n",
      "Epoch 5: val_loss improved from 1.34672 to 1.33998, saving model to ./checkpoints/unknown_person_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 9.7218 - val_loss: 1.3400\n",
      "Epoch 6/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6993\n",
      "Epoch 6: val_loss did not improve from 1.33998\n",
      "31603/31603 [==============================] - 20s 646us/sample - loss: 9.6993 - val_loss: 1.3604\n",
      "Epoch 7/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6772\n",
      "Epoch 7: val_loss improved from 1.33998 to 1.33882, saving model to ./checkpoints/unknown_person_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 645us/sample - loss: 9.6772 - val_loss: 1.3388\n",
      "Epoch 8/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6598\n",
      "Epoch 8: val_loss did not improve from 1.33882\n",
      "31603/31603 [==============================] - 22s 697us/sample - loss: 9.6598 - val_loss: 1.3497\n",
      "Epoch 9/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6408\n",
      "Epoch 9: val_loss did not improve from 1.33882\n",
      "31603/31603 [==============================] - 21s 672us/sample - loss: 9.6408 - val_loss: 1.3400\n",
      "Epoch 10/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6184\n",
      "Epoch 10: val_loss did not improve from 1.33882\n",
      "31603/31603 [==============================] - 20s 622us/sample - loss: 9.6184 - val_loss: 1.3474\n",
      "Epoch 11/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6053\n",
      "Epoch 11: val_loss improved from 1.33882 to 1.33869, saving model to ./checkpoints/unknown_person_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 691us/sample - loss: 9.6053 - val_loss: 1.3387\n",
      "Epoch 12/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.6020\n",
      "Epoch 12: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 21s 661us/sample - loss: 9.6020 - val_loss: 1.3445\n",
      "Epoch 13/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5835\n",
      "Epoch 13: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 9.5835 - val_loss: 1.3465\n",
      "Epoch 14/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5928\n",
      "Epoch 14: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 9.5928 - val_loss: 1.3527\n",
      "Epoch 15/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5898\n",
      "Epoch 15: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 9.5898 - val_loss: 1.3427\n",
      "Epoch 16/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5400\n",
      "Epoch 16: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 19s 613us/sample - loss: 9.5400 - val_loss: 1.3456\n",
      "Epoch 17/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5275\n",
      "Epoch 17: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 21s 667us/sample - loss: 9.5275 - val_loss: 1.3419\n",
      "Epoch 18/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5394\n",
      "Epoch 18: val_loss did not improve from 1.33869\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 9.5394 - val_loss: 1.3403\n",
      "Epoch 19/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5082\n",
      "Epoch 19: val_loss improved from 1.33869 to 1.33612, saving model to ./checkpoints/unknown_person_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 671us/sample - loss: 9.5082 - val_loss: 1.3361\n",
      "Epoch 20/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5072\n",
      "Epoch 20: val_loss did not improve from 1.33612\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 9.5072 - val_loss: 1.3468\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:53:19.333472: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_17_2/lstm_cell_91/kernel/Assign' id:40758 op device:{requested: '', assigned: ''} def:{{{node lstm_17_2/lstm_cell_91/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_17_2/lstm_cell_91/kernel, lstm_17_2/lstm_cell_91/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:53:21.780956: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21_2/lstm_cell_95/recurrent_kernel/v/Assign' id:45613 op device:{requested: '', assigned: ''} def:{{{node lstm_21_2/lstm_cell_95/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21_2/lstm_cell_95/recurrent_kernel/v, lstm_21_2/lstm_cell_95/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31603 samples, validate on 3526 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:53:26.903222: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:53:39.547814: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3727"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:54:03.050238: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.33890, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 30s 942us/sample - loss: 1.3727 - val_loss: 1.3389\n",
      "Epoch 2/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3726\n",
      "Epoch 2: val_loss improved from 1.33890 to 1.33718, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 1.3726 - val_loss: 1.3372\n",
      "Epoch 3/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3718\n",
      "Epoch 3: val_loss improved from 1.33718 to 1.33717, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 21s 671us/sample - loss: 1.3718 - val_loss: 1.3372\n",
      "Epoch 4/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3703\n",
      "Epoch 4: val_loss improved from 1.33717 to 1.33173, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 693us/sample - loss: 1.3703 - val_loss: 1.3317\n",
      "Epoch 5/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3664\n",
      "Epoch 5: val_loss did not improve from 1.33173\n",
      "31603/31603 [==============================] - 20s 636us/sample - loss: 1.3664 - val_loss: 1.3387\n",
      "Epoch 6/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3665\n",
      "Epoch 6: val_loss did not improve from 1.33173\n",
      "31603/31603 [==============================] - 20s 620us/sample - loss: 1.3665 - val_loss: 1.3333\n",
      "Epoch 7/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3651\n",
      "Epoch 7: val_loss did not improve from 1.33173\n",
      "31603/31603 [==============================] - 20s 617us/sample - loss: 1.3651 - val_loss: 1.3323\n",
      "Epoch 8/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3610\n",
      "Epoch 8: val_loss did not improve from 1.33173\n",
      "31603/31603 [==============================] - 21s 650us/sample - loss: 1.3610 - val_loss: 1.3323\n",
      "Epoch 9/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3616\n",
      "Epoch 9: val_loss did not improve from 1.33173\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 1.3616 - val_loss: 1.3323\n",
      "Epoch 10/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3580\n",
      "Epoch 10: val_loss improved from 1.33173 to 1.32919, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 697us/sample - loss: 1.3580 - val_loss: 1.3292\n",
      "Epoch 11/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3587\n",
      "Epoch 11: val_loss did not improve from 1.32919\n",
      "31603/31603 [==============================] - 22s 681us/sample - loss: 1.3587 - val_loss: 1.3315\n",
      "Epoch 12/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3574\n",
      "Epoch 12: val_loss did not improve from 1.32919\n",
      "31603/31603 [==============================] - 22s 697us/sample - loss: 1.3574 - val_loss: 1.3324\n",
      "Epoch 13/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3565\n",
      "Epoch 13: val_loss improved from 1.32919 to 1.32791, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 687us/sample - loss: 1.3565 - val_loss: 1.3279\n",
      "Epoch 14/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3558\n",
      "Epoch 14: val_loss improved from 1.32791 to 1.32645, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 706us/sample - loss: 1.3558 - val_loss: 1.3264\n",
      "Epoch 15/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3524\n",
      "Epoch 15: val_loss improved from 1.32645 to 1.32483, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 20s 643us/sample - loss: 1.3524 - val_loss: 1.3248\n",
      "Epoch 16/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 16: val_loss did not improve from 1.32483\n",
      "31603/31603 [==============================] - 20s 638us/sample - loss: 1.3508 - val_loss: 1.3295\n",
      "Epoch 17/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3516\n",
      "Epoch 17: val_loss improved from 1.32483 to 1.32408, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 1.3516 - val_loss: 1.3241\n",
      "Epoch 18/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3523\n",
      "Epoch 18: val_loss improved from 1.32408 to 1.32342, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 19s 614us/sample - loss: 1.3523 - val_loss: 1.3234\n",
      "Epoch 19/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 19: val_loss did not improve from 1.32342\n",
      "31603/31603 [==============================] - 22s 695us/sample - loss: 1.3498 - val_loss: 1.3278\n",
      "Epoch 20/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3477\n",
      "Epoch 20: val_loss improved from 1.32342 to 1.32226, saving model to ./checkpoints/unknown_person_baseline_2_p28_20.h5\n",
      "31603/31603 [==============================] - 22s 690us/sample - loss: 1.3477 - val_loss: 1.3223\n",
      "Train on 31603 samples, validate on 3526 samples\n",
      "Epoch 1/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3452\n",
      "Epoch 1: val_loss improved from inf to 1.32397, saving model to ./checkpoints/unknown_person_baseline_p28_21.h5\n",
      "31603/31603 [==============================] - 22s 706us/sample - loss: 1.3452 - val_loss: 1.3240\n",
      "Epoch 2/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3482\n",
      "Epoch 2: val_loss did not improve from 1.32397\n",
      "31603/31603 [==============================] - 22s 707us/sample - loss: 1.3482 - val_loss: 1.3264\n",
      "Epoch 3/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3440\n",
      "Epoch 3: val_loss improved from 1.32397 to 1.32204, saving model to ./checkpoints/unknown_person_baseline_p28_21.h5\n",
      "31603/31603 [==============================] - 23s 718us/sample - loss: 1.3440 - val_loss: 1.3220\n",
      "Epoch 4/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3438\n",
      "Epoch 4: val_loss improved from 1.32204 to 1.32166, saving model to ./checkpoints/unknown_person_baseline_p28_21.h5\n",
      "31603/31603 [==============================] - 22s 711us/sample - loss: 1.3438 - val_loss: 1.3217\n",
      "Epoch 5/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3400\n",
      "Epoch 5: val_loss did not improve from 1.32166\n",
      "31603/31603 [==============================] - 21s 665us/sample - loss: 1.3400 - val_loss: 1.3287\n",
      "Epoch 6/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3435\n",
      "Epoch 6: val_loss improved from 1.32166 to 1.32136, saving model to ./checkpoints/unknown_person_baseline_p28_21.h5\n",
      "31603/31603 [==============================] - 20s 629us/sample - loss: 1.3435 - val_loss: 1.3214\n",
      "Epoch 7/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3396\n",
      "Epoch 7: val_loss did not improve from 1.32136\n",
      "31603/31603 [==============================] - 20s 628us/sample - loss: 1.3396 - val_loss: 1.3235\n",
      "Epoch 8/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3451\n",
      "Epoch 8: val_loss did not improve from 1.32136\n",
      "31603/31603 [==============================] - 21s 664us/sample - loss: 1.3451 - val_loss: 1.3252\n",
      "Epoch 9/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3510\n",
      "Epoch 9: val_loss improved from 1.32136 to 1.31869, saving model to ./checkpoints/unknown_person_baseline_p28_21.h5\n",
      "31603/31603 [==============================] - 22s 686us/sample - loss: 1.3510 - val_loss: 1.3187\n",
      "Epoch 10/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 10: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 1.3498 - val_loss: 1.3228\n",
      "Epoch 11/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3539\n",
      "Epoch 11: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.3539 - val_loss: 1.3208\n",
      "Epoch 12/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 12: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 621us/sample - loss: 1.3508 - val_loss: 1.3221\n",
      "Epoch 13/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3587\n",
      "Epoch 13: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 651us/sample - loss: 1.3587 - val_loss: 1.3253\n",
      "Epoch 14/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3565\n",
      "Epoch 14: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 674us/sample - loss: 1.3565 - val_loss: 1.3234\n",
      "Epoch 15/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4036\n",
      "Epoch 15: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 22s 698us/sample - loss: 1.4036 - val_loss: 1.3338\n",
      "Epoch 16/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3686\n",
      "Epoch 16: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 638us/sample - loss: 1.3686 - val_loss: 1.3375\n",
      "Epoch 17/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3857\n",
      "Epoch 17: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 634us/sample - loss: 1.3857 - val_loss: 1.3413\n",
      "Epoch 18/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3882\n",
      "Epoch 18: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 630us/sample - loss: 1.3882 - val_loss: 1.3384\n",
      "Epoch 19/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3849\n",
      "Epoch 19: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 1.3849 - val_loss: 1.3489\n",
      "Epoch 20/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3839\n",
      "Epoch 20: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 650us/sample - loss: 1.3839 - val_loss: 1.3509\n",
      "Epoch 21/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4003\n",
      "Epoch 21: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 662us/sample - loss: 1.4003 - val_loss: 1.3474\n",
      "Epoch 22/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4094\n",
      "Epoch 22: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 1.4094 - val_loss: 1.3502\n",
      "Epoch 23/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3967\n",
      "Epoch 23: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 628us/sample - loss: 1.3967 - val_loss: 1.3499\n",
      "Epoch 24/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3893\n",
      "Epoch 24: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 639us/sample - loss: 1.3893 - val_loss: 1.3445\n",
      "Epoch 25/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3825\n",
      "Epoch 25: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 645us/sample - loss: 1.3825 - val_loss: 1.3463\n",
      "Epoch 26/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3809\n",
      "Epoch 26: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 628us/sample - loss: 1.3809 - val_loss: 1.3424\n",
      "Epoch 27/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3786\n",
      "Epoch 27: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 620us/sample - loss: 1.3786 - val_loss: 1.3477\n",
      "Epoch 28/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3737\n",
      "Epoch 28: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 652us/sample - loss: 1.3737 - val_loss: 1.3399\n",
      "Epoch 29/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3709\n",
      "Epoch 29: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 653us/sample - loss: 1.3709 - val_loss: 1.3417\n",
      "Epoch 30/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3693\n",
      "Epoch 30: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 634us/sample - loss: 1.3693 - val_loss: 1.3358\n",
      "Epoch 31/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3677\n",
      "Epoch 31: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 1.3677 - val_loss: 1.3479\n",
      "Epoch 32/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3669\n",
      "Epoch 32: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.3669 - val_loss: 1.3367\n",
      "Epoch 33/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3605\n",
      "Epoch 33: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 632us/sample - loss: 1.3605 - val_loss: 1.3360\n",
      "Epoch 34/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3630\n",
      "Epoch 34: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 634us/sample - loss: 1.3630 - val_loss: 1.3308\n",
      "Epoch 35/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3603\n",
      "Epoch 35: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 621us/sample - loss: 1.3603 - val_loss: 1.3314\n",
      "Epoch 36/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3585\n",
      "Epoch 36: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 626us/sample - loss: 1.3585 - val_loss: 1.3347\n",
      "Epoch 37/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3568\n",
      "Epoch 37: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 632us/sample - loss: 1.3568 - val_loss: 1.3282\n",
      "Epoch 38/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3559\n",
      "Epoch 38: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 640us/sample - loss: 1.3559 - val_loss: 1.3321\n",
      "Epoch 39/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3545\n",
      "Epoch 39: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 631us/sample - loss: 1.3545 - val_loss: 1.3311\n",
      "Epoch 40/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3525\n",
      "Epoch 40: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 624us/sample - loss: 1.3525 - val_loss: 1.3332\n",
      "Epoch 41/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3507\n",
      "Epoch 41: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 663us/sample - loss: 1.3507 - val_loss: 1.3295\n",
      "Epoch 42/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3505\n",
      "Epoch 42: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 668us/sample - loss: 1.3505 - val_loss: 1.3265\n",
      "Epoch 43/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3479\n",
      "Epoch 43: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 661us/sample - loss: 1.3479 - val_loss: 1.3259\n",
      "Epoch 44/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3483\n",
      "Epoch 44: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 1.3483 - val_loss: 1.3286\n",
      "Epoch 45/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3469\n",
      "Epoch 45: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3469 - val_loss: 1.3370\n",
      "Epoch 46/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3478\n",
      "Epoch 46: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 20s 638us/sample - loss: 1.3478 - val_loss: 1.3270\n",
      "Epoch 47/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3454\n",
      "Epoch 47: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 21s 662us/sample - loss: 1.3454 - val_loss: 1.3249\n",
      "Epoch 48/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3442\n",
      "Epoch 48: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 22s 692us/sample - loss: 1.3442 - val_loss: 1.3353\n",
      "Epoch 49/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3445\n",
      "Epoch 49: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 22s 708us/sample - loss: 1.3445 - val_loss: 1.3276\n",
      "Epoch 50/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3395\n",
      "Epoch 50: val_loss did not improve from 1.31869\n",
      "31603/31603 [==============================] - 22s 710us/sample - loss: 1.3395 - val_loss: 1.3253\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:17:59.056778: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_4_3/lstm_cell_115/recurrent_kernel/Assign' id:58058 op device:{requested: '', assigned: ''} def:{{{node lstm_4_3/lstm_cell_115/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_4_3/lstm_cell_115/recurrent_kernel, lstm_4_3/lstm_cell_115/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:18:02.344305: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_3/lstm_cell_127/bias/v/Assign' id:64903 op device:{requested: '', assigned: ''} def:{{{node lstm_16_3/lstm_cell_127/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_3/lstm_cell_127/bias/v, lstm_16_3/lstm_cell_127/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:18:05.487728: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1872)\n",
      "(1514, 1872)\n",
      "(1644, 1872)\n",
      "(1764, 1872)\n",
      "(1836, 1872)\n",
      "(1699, 1872)\n",
      "(1369, 1872)\n",
      "(1778, 1872)\n",
      "(1631, 1872)\n",
      "(1704, 1872)\n",
      "(1550, 1872)\n",
      "(1920, 1872)\n",
      "(1739, 1872)\n",
      "(1788, 1872)\n",
      "(1812, 1872)\n",
      "(1788, 1872)\n",
      "(1776, 1872)\n",
      "(946, 1872)\n",
      "(1860, 1872)\n",
      "{1: 6.2394516093666015, 2: 3.5133420863338207, 4: 9.5292151446518, 5: 6.397446833544136, 6: 6.772274024178604, 8: 8.681413065858752, 9: 4.502857203977891, 10: 9.175429917613052, 11: 6.076169822470967, 12: 10.0, 13: 6.419756288656486, 17: 9.631454112261242, 19: 9.12138771602842, 21: 9.441284926210066, 22: 1.0, 25: 8.438741921078599, 26: 6.501576342297582, 27: 2.53352033735597, 29: 1.6538785507812186}\n",
      "Train on 31603 samples, validate on 3526 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:22:07.420812: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5951\n",
      "Epoch 1: val_loss improved from inf to 1.33473, saving model to ./checkpoints/unknown_person_p28_21.h5\n",
      "31603/31603 [==============================] - 29s 903us/sample - loss: 9.5951 - val_loss: 1.3347\n",
      "Epoch 2/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5427\n",
      "Epoch 2: val_loss did not improve from 1.33473\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 9.5427 - val_loss: 1.3480\n",
      "Epoch 3/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5607\n",
      "Epoch 3: val_loss did not improve from 1.33473\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 9.5607 - val_loss: 1.3404\n",
      "Epoch 4/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4548\n",
      "Epoch 4: val_loss did not improve from 1.33473\n",
      "31603/31603 [==============================] - 19s 612us/sample - loss: 9.4548 - val_loss: 1.3364\n",
      "Epoch 5/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4202\n",
      "Epoch 5: val_loss did not improve from 1.33473\n",
      "31603/31603 [==============================] - 19s 612us/sample - loss: 9.4202 - val_loss: 1.3435\n",
      "Epoch 6/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4108\n",
      "Epoch 6: val_loss did not improve from 1.33473\n",
      "31603/31603 [==============================] - 19s 614us/sample - loss: 9.4108 - val_loss: 1.3380\n",
      "Epoch 7/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3814\n",
      "Epoch 7: val_loss improved from 1.33473 to 1.33374, saving model to ./checkpoints/unknown_person_p28_21.h5\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 9.3814 - val_loss: 1.3337\n",
      "Epoch 8/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3462\n",
      "Epoch 8: val_loss did not improve from 1.33374\n",
      "31603/31603 [==============================] - 20s 639us/sample - loss: 9.3462 - val_loss: 1.3399\n",
      "Epoch 9/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3541\n",
      "Epoch 9: val_loss did not improve from 1.33374\n",
      "31603/31603 [==============================] - 22s 704us/sample - loss: 9.3541 - val_loss: 1.3402\n",
      "Epoch 10/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3281\n",
      "Epoch 10: val_loss did not improve from 1.33374\n",
      "31603/31603 [==============================] - 22s 698us/sample - loss: 9.3281 - val_loss: 1.3366\n",
      "Epoch 11/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3164\n",
      "Epoch 11: val_loss did not improve from 1.33374\n",
      "31603/31603 [==============================] - 22s 700us/sample - loss: 9.3164 - val_loss: 1.3342\n",
      "Epoch 12/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3297\n",
      "Epoch 12: val_loss did not improve from 1.33374\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 9.3297 - val_loss: 1.3352\n",
      "Epoch 13/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3063\n",
      "Epoch 13: val_loss did not improve from 1.33374\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 9.3063 - val_loss: 1.3360\n",
      "Epoch 14/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2854\n",
      "Epoch 14: val_loss improved from 1.33374 to 1.32851, saving model to ./checkpoints/unknown_person_p28_21.h5\n",
      "31603/31603 [==============================] - 21s 653us/sample - loss: 9.2854 - val_loss: 1.3285\n",
      "Epoch 15/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2726\n",
      "Epoch 15: val_loss did not improve from 1.32851\n",
      "31603/31603 [==============================] - 20s 618us/sample - loss: 9.2726 - val_loss: 1.3366\n",
      "Epoch 16/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2771\n",
      "Epoch 16: val_loss did not improve from 1.32851\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 9.2771 - val_loss: 1.3336\n",
      "Epoch 17/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2734\n",
      "Epoch 17: val_loss did not improve from 1.32851\n",
      "31603/31603 [==============================] - 20s 635us/sample - loss: 9.2734 - val_loss: 1.3339\n",
      "Epoch 18/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2520\n",
      "Epoch 18: val_loss did not improve from 1.32851\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 9.2520 - val_loss: 1.3316\n",
      "Epoch 19/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2419\n",
      "Epoch 19: val_loss did not improve from 1.32851\n",
      "31603/31603 [==============================] - 22s 699us/sample - loss: 9.2419 - val_loss: 1.3378\n",
      "Epoch 20/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2501\n",
      "Epoch 20: val_loss did not improve from 1.32851\n",
      "31603/31603 [==============================] - 21s 651us/sample - loss: 9.2501 - val_loss: 1.3309\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:29:11.048525: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_2_4/kernel/Assign' id:76572 op device:{requested: '', assigned: ''} def:{{{node conv2d_2_4/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_2_4/kernel, conv2d_2_4/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:29:15.174561: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21_4/lstm_cell_169/recurrent_kernel/v/Assign' id:84371 op device:{requested: '', assigned: ''} def:{{{node lstm_21_4/lstm_cell_169/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21_4/lstm_cell_169/recurrent_kernel/v, lstm_21_4/lstm_cell_169/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31603 samples, validate on 3526 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:29:21.841820: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:29:43.515060: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3509"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:30:05.511888: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.32324, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 31s 991us/sample - loss: 1.3509 - val_loss: 1.3232\n",
      "Epoch 2/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3420\n",
      "Epoch 2: val_loss improved from 1.32324 to 1.32231, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 22s 709us/sample - loss: 1.3420 - val_loss: 1.3223\n",
      "Epoch 3/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3494\n",
      "Epoch 3: val_loss did not improve from 1.32231\n",
      "31603/31603 [==============================] - 22s 695us/sample - loss: 1.3494 - val_loss: 1.3255\n",
      "Epoch 4/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3410\n",
      "Epoch 4: val_loss did not improve from 1.32231\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.3410 - val_loss: 1.3250\n",
      "Epoch 5/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3361\n",
      "Epoch 5: val_loss improved from 1.32231 to 1.32130, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 22s 698us/sample - loss: 1.3361 - val_loss: 1.3213\n",
      "Epoch 6/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3360\n",
      "Epoch 6: val_loss improved from 1.32130 to 1.31794, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 22s 681us/sample - loss: 1.3360 - val_loss: 1.3179\n",
      "Epoch 7/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3382\n",
      "Epoch 7: val_loss improved from 1.31794 to 1.31703, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 20s 620us/sample - loss: 1.3382 - val_loss: 1.3170\n",
      "Epoch 8/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3332\n",
      "Epoch 8: val_loss did not improve from 1.31703\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3332 - val_loss: 1.3187\n",
      "Epoch 9/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3332\n",
      "Epoch 9: val_loss did not improve from 1.31703\n",
      "31603/31603 [==============================] - 20s 617us/sample - loss: 1.3332 - val_loss: 1.3185\n",
      "Epoch 10/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3326\n",
      "Epoch 10: val_loss improved from 1.31703 to 1.31278, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.3326 - val_loss: 1.3128\n",
      "Epoch 11/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3303\n",
      "Epoch 11: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3303 - val_loss: 1.3146\n",
      "Epoch 12/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3303\n",
      "Epoch 12: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 20s 622us/sample - loss: 1.3303 - val_loss: 1.3164\n",
      "Epoch 13/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3287\n",
      "Epoch 13: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3287 - val_loss: 1.3156\n",
      "Epoch 14/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3292\n",
      "Epoch 14: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 21s 666us/sample - loss: 1.3292 - val_loss: 1.3131\n",
      "Epoch 15/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3282\n",
      "Epoch 15: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 23s 714us/sample - loss: 1.3282 - val_loss: 1.3149\n",
      "Epoch 16/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3279\n",
      "Epoch 16: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 21s 679us/sample - loss: 1.3279 - val_loss: 1.3173\n",
      "Epoch 17/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3250\n",
      "Epoch 17: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 19s 615us/sample - loss: 1.3250 - val_loss: 1.3162\n",
      "Epoch 18/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3271\n",
      "Epoch 18: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 20s 640us/sample - loss: 1.3271 - val_loss: 1.3167\n",
      "Epoch 19/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3235\n",
      "Epoch 19: val_loss did not improve from 1.31278\n",
      "31603/31603 [==============================] - 21s 663us/sample - loss: 1.3235 - val_loss: 1.3133\n",
      "Epoch 20/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3238\n",
      "Epoch 20: val_loss improved from 1.31278 to 1.30899, saving model to ./checkpoints/unknown_person_baseline_2_p28_21.h5\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.3238 - val_loss: 1.3090\n",
      "Train on 31603 samples, validate on 3526 samples\n",
      "Epoch 1/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3222\n",
      "Epoch 1: val_loss improved from inf to 1.31181, saving model to ./checkpoints/unknown_person_baseline_p28_22.h5\n",
      "31603/31603 [==============================] - 20s 625us/sample - loss: 1.3222 - val_loss: 1.3118\n",
      "Epoch 2/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3222\n",
      "Epoch 2: val_loss did not improve from 1.31181\n",
      "31603/31603 [==============================] - 19s 614us/sample - loss: 1.3222 - val_loss: 1.3129\n",
      "Epoch 3/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3205\n",
      "Epoch 3: val_loss did not improve from 1.31181\n",
      "31603/31603 [==============================] - 19s 615us/sample - loss: 1.3205 - val_loss: 1.3134\n",
      "Epoch 4/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3187\n",
      "Epoch 4: val_loss improved from 1.31181 to 1.31124, saving model to ./checkpoints/unknown_person_baseline_p28_22.h5\n",
      "31603/31603 [==============================] - 19s 617us/sample - loss: 1.3187 - val_loss: 1.3112\n",
      "Epoch 5/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3157\n",
      "Epoch 5: val_loss did not improve from 1.31124\n",
      "31603/31603 [==============================] - 19s 612us/sample - loss: 1.3157 - val_loss: 1.3132\n",
      "Epoch 6/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3168\n",
      "Epoch 6: val_loss improved from 1.31124 to 1.31046, saving model to ./checkpoints/unknown_person_baseline_p28_22.h5\n",
      "31603/31603 [==============================] - 22s 693us/sample - loss: 1.3168 - val_loss: 1.3105\n",
      "Epoch 7/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3140\n",
      "Epoch 7: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 21s 679us/sample - loss: 1.3140 - val_loss: 1.3143\n",
      "Epoch 8/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3294\n",
      "Epoch 8: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 21s 672us/sample - loss: 1.3294 - val_loss: 1.3136\n",
      "Epoch 9/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3360\n",
      "Epoch 9: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 683us/sample - loss: 1.3360 - val_loss: 1.3172\n",
      "Epoch 10/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3293\n",
      "Epoch 10: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 1.3293 - val_loss: 1.3284\n",
      "Epoch 11/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3429\n",
      "Epoch 11: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 705us/sample - loss: 1.3429 - val_loss: 1.3287\n",
      "Epoch 12/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3522\n",
      "Epoch 12: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 709us/sample - loss: 1.3522 - val_loss: 1.3312\n",
      "Epoch 13/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3496\n",
      "Epoch 13: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 706us/sample - loss: 1.3496 - val_loss: 1.3235\n",
      "Epoch 14/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3524\n",
      "Epoch 14: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 1.3524 - val_loss: 1.3243\n",
      "Epoch 15/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3731\n",
      "Epoch 15: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 705us/sample - loss: 1.3731 - val_loss: 1.3353\n",
      "Epoch 16/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3951\n",
      "Epoch 16: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 1.3951 - val_loss: 1.3442\n",
      "Epoch 17/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3663\n",
      "Epoch 17: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 624us/sample - loss: 1.3663 - val_loss: 1.3339\n",
      "Epoch 18/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3798\n",
      "Epoch 18: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 622us/sample - loss: 1.3798 - val_loss: 1.3345\n",
      "Epoch 19/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3665\n",
      "Epoch 19: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 646us/sample - loss: 1.3665 - val_loss: 1.3370\n",
      "Epoch 20/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3876\n",
      "Epoch 20: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 633us/sample - loss: 1.3876 - val_loss: 1.3423\n",
      "Epoch 21/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4663\n",
      "Epoch 21: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 627us/sample - loss: 1.4663 - val_loss: 1.3625\n",
      "Epoch 22/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4669\n",
      "Epoch 22: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 645us/sample - loss: 1.4669 - val_loss: 1.3616\n",
      "Epoch 23/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4211\n",
      "Epoch 23: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 1.4211 - val_loss: 1.3571\n",
      "Epoch 24/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.4016\n",
      "Epoch 24: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 620us/sample - loss: 1.4016 - val_loss: 1.3562\n",
      "Epoch 25/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3955\n",
      "Epoch 25: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 623us/sample - loss: 1.3955 - val_loss: 1.3584\n",
      "Epoch 26/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3899\n",
      "Epoch 26: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 1.3899 - val_loss: 1.3536\n",
      "Epoch 27/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3934\n",
      "Epoch 27: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 618us/sample - loss: 1.3934 - val_loss: 1.3539\n",
      "Epoch 28/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3926\n",
      "Epoch 28: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 626us/sample - loss: 1.3926 - val_loss: 1.3539\n",
      "Epoch 29/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3907\n",
      "Epoch 29: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 1.3907 - val_loss: 1.3530\n",
      "Epoch 30/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3846\n",
      "Epoch 30: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 710us/sample - loss: 1.3846 - val_loss: 1.3554\n",
      "Epoch 31/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3814\n",
      "Epoch 31: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 21s 678us/sample - loss: 1.3814 - val_loss: 1.3528\n",
      "Epoch 32/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3844\n",
      "Epoch 32: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 706us/sample - loss: 1.3844 - val_loss: 1.3538\n",
      "Epoch 33/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3831\n",
      "Epoch 33: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 710us/sample - loss: 1.3831 - val_loss: 1.3438\n",
      "Epoch 34/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3803\n",
      "Epoch 34: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 709us/sample - loss: 1.3803 - val_loss: 1.3449\n",
      "Epoch 35/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3843\n",
      "Epoch 35: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 21s 662us/sample - loss: 1.3843 - val_loss: 1.3477\n",
      "Epoch 36/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3806\n",
      "Epoch 36: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 637us/sample - loss: 1.3806 - val_loss: 1.3541\n",
      "Epoch 37/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3771\n",
      "Epoch 37: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 1.3771 - val_loss: 1.3404\n",
      "Epoch 38/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3746\n",
      "Epoch 38: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 705us/sample - loss: 1.3746 - val_loss: 1.3464\n",
      "Epoch 39/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3749\n",
      "Epoch 39: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 1.3749 - val_loss: 1.3420\n",
      "Epoch 40/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3697\n",
      "Epoch 40: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 690us/sample - loss: 1.3697 - val_loss: 1.3389\n",
      "Epoch 41/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3951\n",
      "Epoch 41: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 644us/sample - loss: 1.3951 - val_loss: 1.3489\n",
      "Epoch 42/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3890\n",
      "Epoch 42: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 689us/sample - loss: 1.3890 - val_loss: 1.3450\n",
      "Epoch 43/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3795\n",
      "Epoch 43: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 627us/sample - loss: 1.3795 - val_loss: 1.3395\n",
      "Epoch 44/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3723\n",
      "Epoch 44: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 626us/sample - loss: 1.3723 - val_loss: 1.3406\n",
      "Epoch 45/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3698\n",
      "Epoch 45: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 710us/sample - loss: 1.3698 - val_loss: 1.3370\n",
      "Epoch 46/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3679\n",
      "Epoch 46: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 23s 713us/sample - loss: 1.3679 - val_loss: 1.3357\n",
      "Epoch 47/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3684\n",
      "Epoch 47: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 634us/sample - loss: 1.3684 - val_loss: 1.3381\n",
      "Epoch 48/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3652\n",
      "Epoch 48: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 20s 632us/sample - loss: 1.3652 - val_loss: 1.3307\n",
      "Epoch 49/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3635\n",
      "Epoch 49: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 22s 704us/sample - loss: 1.3635 - val_loss: 1.3388\n",
      "Epoch 50/50\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3611\n",
      "Epoch 50: val_loss did not improve from 1.31046\n",
      "31603/31603 [==============================] - 21s 661us/sample - loss: 1.3611 - val_loss: 1.3381\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:54:15.939840: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21_5/lstm_cell_206/bias/Assign' id:99545 op device:{requested: '', assigned: ''} def:{{{node lstm_21_5/lstm_cell_206/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21_5/lstm_cell_206/bias, lstm_21_5/lstm_cell_206/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:54:21.108578: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_8_5/lstm_cell_193/kernel/m/Assign' id:102888 op device:{requested: '', assigned: ''} def:{{{node lstm_8_5/lstm_cell_193/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_8_5/lstm_cell_193/kernel/m, lstm_8_5/lstm_cell_193/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:54:26.077338: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1872)\n",
      "(1514, 1872)\n",
      "(1644, 1872)\n",
      "(1764, 1872)\n",
      "(1836, 1872)\n",
      "(1699, 1872)\n",
      "(1369, 1872)\n",
      "(1778, 1872)\n",
      "(1631, 1872)\n",
      "(1704, 1872)\n",
      "(1550, 1872)\n",
      "(1920, 1872)\n",
      "(1739, 1872)\n",
      "(1788, 1872)\n",
      "(1812, 1872)\n",
      "(1788, 1872)\n",
      "(1776, 1872)\n",
      "(946, 1872)\n",
      "(1860, 1872)\n",
      "{1: 6.059806996783478, 2: 3.472146803458669, 4: 9.774244659329193, 5: 6.875215809493295, 6: 6.658767895289443, 8: 8.716009297600026, 9: 4.776574716983604, 10: 9.183706807579819, 11: 6.158836084836134, 12: 10.0, 13: 6.640359397621286, 17: 9.73044374421952, 19: 9.269786432721931, 21: 9.555491998001362, 22: 1.0, 25: 8.44114140368033, 26: 6.457552751611983, 27: 2.7443264212756153, 29: 1.7209141806266648}\n",
      "Train on 31603 samples, validate on 3526 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:58:35.708986: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5706\n",
      "Epoch 1: val_loss improved from inf to 1.35368, saving model to ./checkpoints/unknown_person_p28_22.h5\n",
      "31603/31603 [==============================] - 34s 1ms/sample - loss: 9.5706 - val_loss: 1.3537\n",
      "Epoch 2/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5041\n",
      "Epoch 2: val_loss improved from 1.35368 to 1.33670, saving model to ./checkpoints/unknown_person_p28_22.h5\n",
      "31603/31603 [==============================] - 21s 673us/sample - loss: 9.5041 - val_loss: 1.3367\n",
      "Epoch 3/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.5786\n",
      "Epoch 3: val_loss did not improve from 1.33670\n",
      "31603/31603 [==============================] - 21s 665us/sample - loss: 9.5786 - val_loss: 1.3474\n",
      "Epoch 4/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4780\n",
      "Epoch 4: val_loss improved from 1.33670 to 1.32255, saving model to ./checkpoints/unknown_person_p28_22.h5\n",
      "31603/31603 [==============================] - 21s 660us/sample - loss: 9.4780 - val_loss: 1.3226\n",
      "Epoch 5/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4529\n",
      "Epoch 5: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 696us/sample - loss: 9.4529 - val_loss: 1.3358\n",
      "Epoch 6/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4187\n",
      "Epoch 6: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 688us/sample - loss: 9.4187 - val_loss: 1.3366\n",
      "Epoch 7/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4108\n",
      "Epoch 7: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 9.4108 - val_loss: 1.3239\n",
      "Epoch 8/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.4064\n",
      "Epoch 8: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 9.4064 - val_loss: 1.3280\n",
      "Epoch 9/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3647\n",
      "Epoch 9: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 9.3647 - val_loss: 1.3272\n",
      "Epoch 10/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3716\n",
      "Epoch 10: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 9.3716 - val_loss: 1.3229\n",
      "Epoch 11/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3551\n",
      "Epoch 11: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 701us/sample - loss: 9.3551 - val_loss: 1.3276\n",
      "Epoch 12/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3743\n",
      "Epoch 12: val_loss did not improve from 1.32255\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 9.3743 - val_loss: 1.3479\n",
      "Epoch 13/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3559\n",
      "Epoch 13: val_loss improved from 1.32255 to 1.32206, saving model to ./checkpoints/unknown_person_p28_22.h5\n",
      "31603/31603 [==============================] - 20s 629us/sample - loss: 9.3559 - val_loss: 1.3221\n",
      "Epoch 14/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3282\n",
      "Epoch 14: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 20s 628us/sample - loss: 9.3282 - val_loss: 1.3224\n",
      "Epoch 15/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3305\n",
      "Epoch 15: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 19s 608us/sample - loss: 9.3305 - val_loss: 1.3357\n",
      "Epoch 16/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3201\n",
      "Epoch 16: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 9.3201 - val_loss: 1.3251\n",
      "Epoch 17/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3026\n",
      "Epoch 17: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 22s 705us/sample - loss: 9.3026 - val_loss: 1.3336\n",
      "Epoch 18/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3170\n",
      "Epoch 18: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 9.3170 - val_loss: 1.3237\n",
      "Epoch 19/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.3103\n",
      "Epoch 19: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 9.3103 - val_loss: 1.3249\n",
      "Epoch 20/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 9.2948\n",
      "Epoch 20: val_loss did not improve from 1.32206\n",
      "31603/31603 [==============================] - 22s 700us/sample - loss: 9.2948 - val_loss: 1.3330\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 22:05:59.690751: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_33_6/lstm_cell_255/kernel/Assign' id:120836 op device:{requested: '', assigned: ''} def:{{{node lstm_33_6/lstm_cell_255/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_33_6/lstm_cell_255/kernel, lstm_33_6/lstm_cell_255/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 22:06:05.695523: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_3_6/lstm_cell_225/recurrent_kernel/v/Assign' id:122859 op device:{requested: '', assigned: ''} def:{{{node lstm_3_6/lstm_cell_225/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_3_6/lstm_cell_225/recurrent_kernel/v, lstm_3_6/lstm_cell_225/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31603 samples, validate on 3526 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 22:06:14.651846: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 22:06:46.266684: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3155"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 22:07:07.952040: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.31350, saving model to ./checkpoints/unknown_person_baseline_2_p28_22.h5\n",
      "31603/31603 [==============================] - 49s 2ms/sample - loss: 1.3155 - val_loss: 1.3135\n",
      "Epoch 2/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3118\n",
      "Epoch 2: val_loss did not improve from 1.31350\n",
      "31603/31603 [==============================] - 22s 703us/sample - loss: 1.3118 - val_loss: 1.3194\n",
      "Epoch 3/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3283\n",
      "Epoch 3: val_loss improved from 1.31350 to 1.30968, saving model to ./checkpoints/unknown_person_baseline_2_p28_22.h5\n",
      "31603/31603 [==============================] - 22s 686us/sample - loss: 1.3283 - val_loss: 1.3097\n",
      "Epoch 4/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3199\n",
      "Epoch 4: val_loss did not improve from 1.30968\n",
      "31603/31603 [==============================] - 21s 678us/sample - loss: 1.3199 - val_loss: 1.3108\n",
      "Epoch 5/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3191\n",
      "Epoch 5: val_loss did not improve from 1.30968\n",
      "31603/31603 [==============================] - 21s 670us/sample - loss: 1.3191 - val_loss: 1.3137\n",
      "Epoch 6/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3155\n",
      "Epoch 6: val_loss improved from 1.30968 to 1.30938, saving model to ./checkpoints/unknown_person_baseline_2_p28_22.h5\n",
      "31603/31603 [==============================] - 20s 648us/sample - loss: 1.3155 - val_loss: 1.3094\n",
      "Epoch 7/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3158\n",
      "Epoch 7: val_loss improved from 1.30938 to 1.30801, saving model to ./checkpoints/unknown_person_baseline_2_p28_22.h5\n",
      "31603/31603 [==============================] - 22s 695us/sample - loss: 1.3158 - val_loss: 1.3080\n",
      "Epoch 8/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3172\n",
      "Epoch 8: val_loss improved from 1.30801 to 1.30678, saving model to ./checkpoints/unknown_person_baseline_2_p28_22.h5\n",
      "31603/31603 [==============================] - 22s 708us/sample - loss: 1.3172 - val_loss: 1.3068\n",
      "Epoch 9/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3127\n",
      "Epoch 9: val_loss did not improve from 1.30678\n",
      "31603/31603 [==============================] - 22s 694us/sample - loss: 1.3127 - val_loss: 1.3076\n",
      "Epoch 10/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3138\n",
      "Epoch 10: val_loss did not improve from 1.30678\n",
      "31603/31603 [==============================] - 22s 694us/sample - loss: 1.3138 - val_loss: 1.3093\n",
      "Epoch 11/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3139\n",
      "Epoch 11: val_loss did not improve from 1.30678\n",
      "31603/31603 [==============================] - 20s 643us/sample - loss: 1.3139 - val_loss: 1.3091\n",
      "Epoch 12/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3094\n",
      "Epoch 12: val_loss did not improve from 1.30678\n",
      "31603/31603 [==============================] - 22s 702us/sample - loss: 1.3094 - val_loss: 1.3092\n",
      "Epoch 13/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3120\n",
      "Epoch 13: val_loss did not improve from 1.30678\n",
      "31603/31603 [==============================] - 21s 668us/sample - loss: 1.3120 - val_loss: 1.3159\n",
      "Epoch 14/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3119\n",
      "Epoch 14: val_loss did not improve from 1.30678\n",
      "31603/31603 [==============================] - 19s 613us/sample - loss: 1.3119 - val_loss: 1.3161\n",
      "Epoch 15/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3079\n",
      "Epoch 15: val_loss improved from 1.30678 to 1.30519, saving model to ./checkpoints/unknown_person_baseline_2_p28_22.h5\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3079 - val_loss: 1.3052\n",
      "Epoch 16/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3086\n",
      "Epoch 16: val_loss did not improve from 1.30519\n",
      "31603/31603 [==============================] - 19s 616us/sample - loss: 1.3086 - val_loss: 1.3090\n",
      "Epoch 17/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3094\n",
      "Epoch 17: val_loss did not improve from 1.30519\n",
      "31603/31603 [==============================] - 19s 615us/sample - loss: 1.3094 - val_loss: 1.3102\n",
      "Epoch 18/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3062\n",
      "Epoch 18: val_loss did not improve from 1.30519\n",
      "31603/31603 [==============================] - 19s 615us/sample - loss: 1.3062 - val_loss: 1.3086\n",
      "Epoch 19/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3048\n",
      "Epoch 19: val_loss did not improve from 1.30519\n",
      "31603/31603 [==============================] - 20s 619us/sample - loss: 1.3048 - val_loss: 1.3090\n",
      "Epoch 20/20\n",
      "31603/31603 [==============================] - ETA: 0s - loss: 1.3061\n",
      "Epoch 20: val_loss did not improve from 1.30519\n",
      "31603/31603 [==============================] - 20s 618us/sample - loss: 1.3061 - val_loss: 1.3139\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [28]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
